source/extensions/common/aws/signer_impl.cc

```diff
@@ -24,6 +24,9 @@ void SignerImpl::sign(Http::RequestMessage& message, bool sign_body) {
 }
 
 void SignerImpl::sign(Http::RequestHeaderMap& headers) {
+  // We are signing something without a body, so insert the empty string hash.
+  headers.addCopy(SignatureHeaders::get().ContentSha256,
+                  SignatureConstants::get().HashedEmptyString);
```

Comment:
This is one of the areas where s3 is different from all other aws services when it comes to signing. This needs to be the literal `UNSIGNED_PAYLOAD` (only for s3)

Reply:
Note that the above works too.... But sure.

Reply:
It might work for GETs, but does it work for PUTs/uploading? I'd be surprised.

Reply:
i only tried GETs